src/components/WebComponentProject/WebComponentProject.jsx

```diff
@@ -57,14 +62,14 @@ const WebComponentProject = ({
       const mz_criteria = Sk.sense_hat
         ? Sk.sense_hat.mz_criteria
         : { ...defaultMZCriteria };
-      document.dispatchEvent(
-        runCompletedEvent({
-          isErrorFree: error === "",
-          ...mz_criteria,
-        }),
-      );
+
+      const payload = outputOnly
+        ? { errorDetails }
+        : { isErrorFree: error === "", ...mz_criteria };
+
+      document.dispatchEvent(runCompletedEvent(payload));
     }
-  }, [codeRunTriggered, codeHasRun, error]);
+  }, [codeRunTriggered, codeHasRun]);
```

Comment:
Hmm, see the GithubActions linter has raised a concern about the missing dependencies here